BIG_BOT/src/fsm/sequences/sequenceManager.py

```diff
@@ -0,0 +1,96 @@
+from ..commands.command import ICommand
+from ...constants import StateEnum
+from ..myTimer import MyTimer
+import time
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from ..FSM import RobotFSM
+
+class SequenceManager():
+    def __init__(self, fsm: 'RobotFSM', sequences: list[list[ICommand]]) -> None:
+        self.fsm = fsm
+        self._sequences = sequences
+        self._current_sequence: list[ICommand] = self._sequences[0]
+        self._current_sequence_idx: int = 0
+        self._current_command_idx: int = 0
+        self._command: ICommand | None = None
+        self._timer: MyTimer | None = None
+        self._execution_in_progress: bool = False
+        self._all_sequences_completed: bool = False
+
+    def execute_step(self):
+        # First check if all sequences are completed
+        if self._all_sequences_completed:
+            return
+
+        # Don't do anything if we're already executing a command
+        if self._execution_in_progress:
+            return
+
+        # Check if we've finished the current sequence
+        if self._current_command_idx >= len(self._current_sequence):
+            self._current_sequence_idx += 1
+            self._current_command_idx = 0
+            
+            # Check if we've finished all sequences
+            if self._current_sequence_idx >= len(self._sequences):
+                print("All sequences completed")
+                self._all_sequences_completed = True
+                return
+                
+            self._current_sequence = self._sequences[self._current_sequence_idx]
+
+        # Start next command
+        self._command = self._current_sequence[self._current_command_idx]
+        
+        # Set the execution flag to prevent duplicate calls
+        self._execution_in_progress = True
+        
+        # Execute command and get timer
+        time_needed = self._command.execute()
+        self._timer = MyTimer(time_needed, self._on_step_complete)
```

Comment:
I see why you want to get the time needed to perform or finish a command, but it should not be done by returning that time in the `execute` method.
Even when following the Command design pattern, an execute function executes code and that's all.
We also do not need to create a `Timer` for every command. It result in some unnecessary delays and memory usage.

If some commands needs to track the remaining time until it is completed, it should be implemented in the command as an attribute and creating a getter to access it outside.
Commands that do not need these delays will not have that attribute so we could create a Timer only if the `time_needed` getter is present:
```suggestion
        self._command.execute()
        try:
            self._timer = MyTimer(self._command.time_needed, self._on_step_complete)
        except AttributeError:
            self._timer = None  # If no property, do not create Timer
```